File: backend-python/app/routes/analytics.py
"""Analytics routes for tracking frontend events."""

from typing import Optional

from fastapi import APIRouter
from loguru import logger
from pydantic import BaseModel

router = APIRouter()


class AnalyticsEvent(BaseModel):
    """Single analytics event."""

    name: str
    properties: Optional[dict] = None
    timestamp: str


class AnalyticsPayload(BaseModel):
    """Analytics payload with session and events."""

    sessionId: str
    events: list[AnalyticsEvent]


@router.post("/events")
async def receive_events(payload: AnalyticsPayload):
    """
    Receive analytics events from the frontend.

    Currently logs events for debugging. Can be extended to store
    in database or forward to analytics service.
    """
    logger.debug(
        f"Analytics: session {payload.sessionId[:8]}, "
        f"events: {[e.name for e in payload.events]}"
    )

    # In production, you could:
    # - Store in database for analysis
    # - Forward to analytics service (PostHog, Amplitude, etc.)
    # - Aggregate metrics

    return {"received": len(payload.events)}

File: backend/src/routes/analytics.ts
import { Router, Request, Response } from 'express';

const router = Router();

interface AnalyticsEvent {
  name: string;
  properties?: Record<string, string | number | boolean>;
  timestamp: string;
}

interface AnalyticsPayload {
  sessionId: string;
  events: AnalyticsEvent[];
}

// POST /analytics/events - Receive analytics events
router.post('/events', (req: Request, res: Response) => {
  try {
    const payload = req.body as AnalyticsPayload;

    // Validate payload
    if (!payload.sessionId || !Array.isArray(payload.events)) {
      res.status(400).json({ error: 'Invalid payload' });
      return;
    }

    // Log events for now (can add database storage later)
    if (process.env.NODE_ENV !== 'production') {
      console.log(`[Analytics] Session ${payload.sessionId.slice(0, 8)}:`,
        payload.events.map(e => e.name).join(', '));
    }

    // In production, you could store these in a database or forward to an analytics service
    // For now, just acknowledge receipt
    res.status(200).json({ received: payload.events.length });
  } catch (error) {
    console.error('Analytics error:', error);
    res.status(500).json({ error: 'Failed to process analytics' });
  }
});

export default router;
